src/python/WMCore/Database/MongoDB.py

```diff
@@ -44,19 +45,40 @@ def __init__(self, database=None, server=None, port=None, replicaset=None,
             msg = "You are trying to mock MongoDB, but you do not have mongomock in the python path."
             self.logger.critical(msg)
             raise ImportError(msg)
+
+        # NOTE: We need to explicitely check for server availiability.
+        #       From pymongo Documentation:
+        #       """
+        #           ...
+        #           Starting with version 3.0 the :class:`MongoClient`
+        #           constructor no longer blocks while connecting to the server or
+        #           servers, and it no longer raises
+        #           :class:`~pymongo.errors.ConnectionFailure` if they are
+        #           unavailable, nor :class:`~pymongo.errors.ConfigurationError`
+        #           if the user's credentials are wrong. Instead, the constructor
+        #           returns immediately and launches the connection process on
+        #           background threads.
+        #           ...
+        #       """
         try:
             if mockMongoDB:
                 self.client = mongomock.MongoClient()
                 self.logger.info("NOTICE: MongoDB is set to use mongomock, instead of real database.")
             elif replicaset:
-                self.client = MongoClient(self.server, self.port, replicaset=replicaset, **kwargs )
+                self.client = MongoClient(host=self.server, port=self.port, replicaset=replicaset, **kwargs )
```

Comment:
Given that we have the port number as part of the server url (list of urls), does it make any sense to provide a fixed `self.port` as part of the connection string?

Reply:
Yes, it makes sense. For MongoDB there are three types of endpoints, which are intermixable: `URI`, `hostName/ip address` and `mongos`. And they have different ways to provide the set of parameters. Now saying that here is our list of reasons why we  should preserve the port in the connection parameter list:
* we should stay backwords compatible with what we already have in the system - the production server is still using a `URI` to connect to the server and a  single port. And if we fail to reconfigure (in case it is ever needed) the production server, we will have to use that one for  a little longer.
* We would not want to strip our wrapper from the standard set of parameters from the MongoClient constructor signature.